src/background/logger.js

```diff
@@ -0,0 +1,42 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import {Component} from './component.js';
+
+let self;
+
+export class Logger extends Component {
```

Comment:
I do not think this needs to be a class: `logger()`is just a wrapper for loginternal which is a static function itself.
This could be just 2 functions with one exported imho, fine to keep it tho :D